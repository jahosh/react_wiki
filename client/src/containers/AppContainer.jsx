import React, { Component } from 'react';

import Header from '../components/Header';
import SearchBox from '../components/SearchBox';
import CategoryList from '../components/CategoryList';

import { buildArr } from '../utils/helpers';

class AppContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      queryObj: null,
      results: [],
    };
    this.handleSearch = this.handleSearch.bind(this);
  }

  handleSearch(e) {
    const query = e.target.value;
    if (query.length < 2) {
      return;
    }

    fetch(`https://en.wikipedia.org/w/api.php?action=query&format=json&origin=*&generator=allcategories&gacprefix=${query}&prop=info`)
    .then((resp) => {
      return resp.json();
    })
    .then((result) => {
      this.setState({ 
        queryObj: result,
        results: buildArr(result.query.pages)
      });
    })
    .catch(err => console.log(err));
  }

  render() {
    console.log(this.state);
    return (
      <div className="ui grid">
        <div className="row">
          <div className="ui four column centered grid">
            <Header />
          </div>
        </div>

        <div className="centered row">
          <div className="ui four colum wide">
            <SearchBox
              handleSearch={this.handleSearch}
            />
          </div>
        </div>

        <div className="centered row">
          <div className="ui eight colum wide">
            { this.state.results.length ?
              <CategoryList
                categories={this.state.results}
              />
              :
              ''
            }
          </div>
        </div>

      </div>
    );
  }
}

export default AppContainer;
