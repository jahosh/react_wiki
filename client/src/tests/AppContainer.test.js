import React from 'react';
import renderer from 'react-test-renderer';
import AppContainer from '../containers/AppContainer';

test('AppContainer renders properly', () => {
  const component = renderer.create(
    <AppContainer />
  )

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
