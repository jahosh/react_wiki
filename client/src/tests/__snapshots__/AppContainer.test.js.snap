// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContainer renders properly 1`] = `
<div
  className="ui grid container centered"
>
  <div
    className="row"
  >
    <div
      className="ui eight column "
    >
      <div
        id="header"
      >
        <h1>
           React-Wiki
        </h1>
        <div
          className="sub-header"
        >
          Josh Hickman's front-end candidate submission
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="ui eight colum "
    >
      <div
        className="ui huge icon input"
      >
        <input
          onChange={[Function]}
          onKeyUp={[Function]}
          placeholder="Search Wikipedia..."
          type="text"
        />
        <i
          className="inverted circular search link icon"
        />
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="ui eight colum"
    >
      
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="ui sixteen colum"
    >
      
    </div>
  </div>
</div>
`;
